src/core/src/package_managers/YumPackageManager.py

```diff
@@ -325,26 +325,126 @@ def extract_dependencies(self, output, packages):
+        # Upgrading:
+        # kernel-tools        x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms  8.4 M
+        # kernel-tools-libs   x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms  8.2 M
+        # openssl             x86_64  1:1.1.1k-9.el8_6      rhel-8-for-x86_64-baseos-eus-rhui-rpms  710 k
+        # openssl-libs        x86_64  1:1.1.1k-9.el8_6      rhel-8-for-x86_64-baseos-eus-rhui-rpms  1.5 M
+        # Installing dependencies:
+        # kernel-core         x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms   40 M
+        # kernel-modules      x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms   32 M
+        #
+        # Transaction Summary
+        # ================================================================================================
+        # Install  2 Packages
+        # Upgrade  4 Packages
+
+        # In Yum 4: Sample 4 (dependent patch detail split over 2 lines):
+        # Sample output for the cmd 'sudo yum update --assumeno polkit.x86_64' is :
+        #
+        # Last metadata expiration check: 0:08:47 ago on Tue 25 Jul 2023 02:14:28 PM UTC.
+        # Package polkit-0.115-13.el8_5.2.x86_64 is already installed.
+        # Dependencies resolved.
+        # ================================================================================
+        # Package   Arch   Version          Repository                              Size
+        # ================================================================================
+        # Upgrading:
+        # polkit    x86_64 0.115-14.el8_6.1 rhel-8-for-x86_64-baseos-eus-rhui-rpms 154 k
+        # polkit-libs
+        #           x86_64 0.115-14.el8_6.1 rhel-8-for-x86_64-baseos-eus-rhui-rpms  77 k
+        #
+        # Transaction Summary
+        # ================================================================================
+        # Upgrade  2 Packages
+        #
+        # Total download size: 231 k
+        # Operation aborted.
+
         dependencies = []
+        package_arch_to_look_for = ["x86_64", "noarch", "i686"]
+
         lines = output.strip().split('\n')
 
-        for line in lines:
-            if line.find(" will be updated") < 0 and line.find(" will be an update") < 0 and line.find(" will be installed") < 0:
-                self.composite_logger.log_debug(" - Inapplicable line: " + str(line))
-                continue
+        for line_index in range(0, len(lines)):
+            line = re.split(r'\s+', (lines[line_index].replace("--->", "")).strip())
+            next_line = []
+            dependent_package_name = ""
+
+            if line_index < len(lines) - 1:
+                next_line = re.split(r'\s+', (lines[line_index + 1].replace("--->", "")).strip())
 
-            updates_line = re.split(r'\s+', line.strip())
-            if len(updates_line) != 7:
+            if self.is_valid_update(line, package_arch_to_look_for):
```

Comment:
we can combine two ifs into one
newLine = line if not next_line else line+new_line

if self.is_valid_update(newLine, package_arch_to_look_for):
dependent_package_name = self.get_product_name_with_arch(newLine, package_arch_to_look_for)

Reply:
Not sure if I fully follow this comment, which are the 2 ifs you are considering here? 

The one starting with if self.is_valid_update(newLine, package_arch_to_look_for): is a complete if, elif else block, that cannot be combined with if line_index < len(lines) - 1: on line 423

Reply:
if self.is_valid_update(line, package_arch_to_look_for):
                dependent_package_name = self.get_product_name_with_arch(line, package_arch_to_look_for)
elif self.is_valid_update(line+next_line, package_arch_to_look_for):
                dependent_package_name = self.get_product_name_with_arch(line+next_line, package_arch_to_look_for)
 --->         
 newLine = line if not next_line else line+new_line
 if self.is_valid_update(newLine, package_arch_to_look_for):
     dependent_package_name = self.get_product_name_with_arch(newLine, package_arch_to_look_for)